sqld/src/lib.rs

```diff
@@ -228,6 +237,53 @@ fn check_fresh_db(path: &Path) -> bool {
     !path.join("wallog").exists()
 }
 
+fn validate_extensions(extensions_path: Option<PathBuf>) -> anyhow::Result<Vec<PathBuf>> {
+    let mut valid_extensions = vec![];
+    if let Some(ext_dir) = extensions_path {
+        let extensions_list = ext_dir.join("trusted.lst");
+        let mut list = File::open(&extensions_list).with_context(|| {
+            format!(
+                "While trying to read extensions: cannot load {}",
+                &extensions_list.display()
+            )
+        })?; // anyhow;
+        let mut file_contents = String::new();
+
+        list.read_to_string(&mut file_contents)
+            .with_context(|| format!("can't read {}", &extensions_list.display()))?;
```

Comment:
You can use `std::fs::read_to_string()` to open the file and read its content in a single step, something like `let file_contents = std::fs::read_to_string(&extensions_list).with_context(|| { ... })`